apps/fastlane/helpers/version_helper.rb

```diff
@@ -13,14 +13,30 @@
 
   # Replace '/' with '-' to avoid issues with unsupported characters in version name
   branch_name = branch_name.gsub('/', '-')
-  timestamp = current_time.strftime("%Y%m%d.%H%M%S")
+  # Extract ticket number from branch name
+  ticket_number_in_branch_name = branch_name.scan(/\d+/).join
+  # If no ticket number found, set it to 0
+  ticket_number_in_branch_name = ticket_number_in_branch_name.empty? ? "0" : ticket_number_in_branch_name
 
-  sdk_version_name = "#{timestamp}.0-#{branch_name}"
+  # Format the components individually and remove leading zeros
+  year = current_time.year.to_s
+  month = current_time.month.to_s
+  day = current_time.day.to_s
+  hour = current_time.hour.to_s
+  minute = current_time.min.to_s
+  second = current_time.sec.to_s
+
+  # Combine them into the desired format
+  major = "#{year}#{month}#{day}"
+  minor = "#{hour}#{minute}#{second}"
+  patch = ticket_number_in_branch_name.to_s
+
+  sdk_version_name = "#{major}.#{minor}.#{patch}"
```

Comment:
This matches with that we do in [React Native](https://github.com/customerio/customerio-reactnative/blob/e8535a10c0e2d1e07fb2b160cc1d194e54c6e2f3/Apps/fastlane/helpers/version_helper.rb#L5-L48) today

Reply:
not fully sure whats happening here and why its needed, can you add some comments? i understand its the same in react native and we can probably add comments there too. 😕